stingray/tests/test_lightcurve.py

```diff
@@ -876,7 +876,8 @@ def test_rebin_with_gtis(self):
         good = create_gti_mask(times, gti)
 
         counts[np.logical_not(good)] = 0
-        lc = Lightcurve(times, counts, gti=gti)
+        lc = Lightcurve(times, counts, gti=gti, skip_checks=True, dt=0.1)
+        lc._apply_gtis()
```

Comment:
Are there ways to test the lazy evaluations?  I think that would be useful. I think it would also be great to have a test or two for `skip_checks`

Reply:
The lazy evaluations (i.e. all the new "properties") are tested everywhere, because they are run every time we do any operation in the light curve. We call each of them at least once. I'm not sure I know how to test them deeper than that?

Reply:
Ok, I actually added a few test to most of them now, and more checks on data validity (e.g. not having a new count array of with inconsistent dimension etc.)